db/hints/manager.cc

```diff
@@ -1006,5 +1012,19 @@ void manager::update_backlog(size_t backlog, size_t max_backlog) {
     }
 }
 
+void manager::on_host_filter_update(const host_filter& filter) {
+    manager_logger.debug("updating host filtering configuration: {}", filter);
+
+    _host_filter = filter;
+
+    for (auto& [ep, mgr] : _ep_managers) {
+        if (filter.can_hint_for(_local_snitch_ptr, ep)) {
```

Comment:
This is not enough - this will leave manages to "blocked" DCs alive and they will continue sending hints. 
You should prevent them from doing that and you should also stop the corresponding sender thread from waking up every second too.

The cleanest would be to shut them down in the context of this call.

This in particular means that using the updatable_vlaue in its current state is not a good choice because it doesn't allow propagating the state of callbacks to the caller (e.g. `updateable_value_source::set(T) could have returned `future` that would resolve only when all callbacks' returned futures resolve to success and would rollback if any of them fails).

This would create a clean user interface where the caller would know that the change has been actually applied when the corresponding future resolves, for instance when triggered from the REST API. It would also allow letting the user know that the state change has failed.

No matter what however you should still keep the `hints::manager` clean.
Change in hints enable/disable should not leave senders alive and the corresponding call should denote the completion of the operation including a rollback if a state transition has failed.

If in the current state of the `liveness::LiveUpdate` framework it's impossible (as it seems to me) then we should probably not allow changing `hints_enabled` value in the runtime for now and wait till it becomes possible.

Given the state of things today (if I'm not missing something big here) the best you can do it to attempt shutting corresponding endpoint_managers in the background.
However this will be a very confusing interface because there will be no way to determine when the change has actually took place (when new hints are both stopped been generated and old hints being sent to corresponding destinations).

Having said all that I think it makes more sense to start with improving the `liveness::LiveUpdate` framework in order to address what's described above and then get back to this patch and do it properly.

Reply:
This shouldn't be a problem in v3 - changing configuration is now only possible through HTTP, it causes appropriate hint managers to be stopped/started, and the HTTP endpoint for changing configuration is now blocking and waits for the operation to complete.